content/posts/gsoc-physiopy/index.md

```diff
@@ -2,6 +2,104 @@
+Since `phys2denoise`'s use is pretty straightforward, the workflow turned out to be linear, with all component `pydra` tasks, being used sequentially. The following three tasks were defined for the workflow:
+
+{{< mermaid >}}
+graph LR;
+A[transform_to_physio]-->B[compute_metrics];
+B-->C[export_metrics]
+style A fill:#c197db,stroke:#333
+{{< /mermaid >}}
+
+The `transform_to_physio` task is part of `physutils`, as it serves as an interface task to be used in other physiopy workflows (prep4phys, physioqc, etc.), to transform multiple input formats into a `Physio` object, which is the primary I/O object throughout the tasks and functions. 
+Its primary function is to take as an input either a "physio" file of any type that contains 1D data, including pickled `Physio` object files, or a BIDS data structure. Within the workflow the input data type can also be deduced automatically. To accomplish that it uses `physutils.load_physio` and `physutils.load_from_bids`.
+
+```python
+def transform_to_physio(
+    input_file: str, mode="physio", fs=None, 
+    bids_parameters=dict(), bids_channel=None 
+) -> Physio:
+```
+
+The `compute_metrics` task is used to check if all the provided metrics can be computed (i.e. all the required parameters and signal properties are calculated or given), computes them, and then stores them in the `Physio` object for further use.
+It utilizes all the functions from `phys2denoise.metrics` for the actual computations, and a `select_input_args` function to verify that the metric can be computed
+
+```python
+def compute_metrics(
+    phys: Physio, metrics: Union[list, str], 
+    args: dict
+) -> Physio:
+```
+
+Finally, the `export_metrics` task exports the specified metrics, which could be a subset of the computed ones. 
+If no export metric parameters are specified, the CLI defaults to exporting all the metrics. 
+Exporting includes generating 1D files that contain the metric. If a metric contains lags, then a file will be exported for each lag value, and if a metric is convolved it will export both a file of the convolved signal and a file of the raw signal. Furthermore, all metrics are both exported in the original sample rate and resampled at the fMRI scan TR.
+```python
+def export_metrics(
+    phys: Physio, metrics: Union[list, str], 
+    outdir: str, tr: Union[int, float]
+) -> int:
+```
+
+A current version of a CLI run can be seen as captured below
+
+<script src="https://asciinema.org/a/PHi5AIIpxSH9jsq1AvKELom09.js" id="asciicast-PHi5AIIpxSH9jsq1AvKELom09" async="true"></script>
+
+## 4. Future Directions
+
+### 4.1 Infrastructure
+
+### 4.2 CLI for peakdet
+
+### 4.3 CLI for physiopy suite
+
+---
+Lastly, I'd like to express my gratitude towards my mentors Stefano Moia ([@smoia](https://github.com/smoia)), Mary-Eve Picard ([@me-pic](https://github.com/me-pic)) and Mary Miedema ([@m-miedema](https://github.com/m-miedema)) and the physiopy community for providing me with this great experience. 
```

Comment:
We're really grateful to you too, Georgios!! A small note that it's Marie-Eve :)